lib/libmbedtls/core/cmac.c

```diff
@@ -39,6 +39,17 @@ static TEE_Result mbed_cmac_init(struct crypto_mac_ctx *ctx,
 	struct mbed_cmac_ctx *c = to_cmac_ctx(ctx);
 	const mbedtls_cipher_info_t *cipher_info = NULL;
 
+	if (len == 16) {
+		/*
+		 * mbedTLS supports 128 bit keys for MBEDTLS_CIPHER_ID_DES
+		 * and not for MBEDTLS_CIPHER_ID_3DES.
+		 * That's why we have to change the cipher id here.
+		 */
+		if (c->cipher_id == MBEDTLS_CIPHER_ID_3DES) {
+			c->cipher_id = MBEDTLS_CIPHER_ID_DES;
+		}
+	}
+
```

Comment:
I'm not sure I understand all what is happening here, but frankly I don't like this much. You are pretending to use an algorithm when you are using a different one, right? Why not simply create a proper 192-bit key from the 128-bit one and be done with it?

Reply:
Actually I had at least 4 ways how to do it:
1) the most straightforward one:
just extend 128 bit keys here to 192 bits and let it flow
2) mbedTLS in theory has a flag for algorithm that indicates that it support keys of variable length but using this approach will require more changes in mbedTLS core than any of the approaches described here.
3) don't setup cipher in crypto_cmac_alloc_ctx() in a case of 3DES cipher and setup it at mbed_cmac_init() step when key length is known.
This still requires the same changes as proposed in this very PR though.
4) this variant which seems as a good compromise to me as the required changes are minimal compared to (2) and (3), and at the same time we are not required to copy keys here (so we keep the same behaviour as in case if libtomcrypt  is used as a core crypto library)

Reply:
I'd like to see option (1) actually. I wouldn't be surprise it would be the simplest, easiest to understand and to maintain.

Reply:
I'd prefer a single `if ()`:
```C
	if (len == 16 && c->cipher_id == MBEDTLS_CIPHER_ID_3DES) {
		/*
		 * mbedTLS supports 128 bit keys for MBEDTLS_CIPHER_ID_DES
		 * and not for MBEDTLS_CIPHER_ID_3DES.
		 * That's why we have to change the cipher id here.
		 */
		c->cipher_id = MBEDTLS_CIPHER_ID_DES;
	}
```